src/utils/Constraint.js

```diff
@@ -47,6 +47,49 @@ export class DistanceConstraint extends Constraint {
   }
 }
 
+// Describes a constraint in which two particles must be
+export class PearlPearlConstraint extends Constraint {
```

Comment:
very similar to the distance constraint, but it only moves if the distance is less than the sum of the pearl_radii